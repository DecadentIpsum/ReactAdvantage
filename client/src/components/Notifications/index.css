.notification-button {
  background: none;
  border: none;
  cursor: pointer;
  font-size: 1rem;
  color: #5A88C3; }
