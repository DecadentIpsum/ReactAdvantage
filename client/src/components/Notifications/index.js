import React, { Component } from 'react';
import { TransitionGroup } from 'react-transition-group';
import FadeInContainer from 'components/FadeInContainer';
import HideOnClickOutsideContainer from 'components/HideOnClickOutsideContainer';

import './index.css';

export default class Notifications extends Component {
    onNotificationsClick = () => {
        this.togglePopup();
    }

    setTogglerRef = ref => {
        this.togglerRef = ref;
    }

    state = {
        showPopup: false,
    }

    togglePopup = (state = null) => {
        this.setState({
            showPopup: state === null
                ? !this.state.showPopup
                : state
        });
    }

    renderNotifications() {
        return (
            <FadeInContainer>
                <div className="notifications-popup">
                    <div className="notifications-popup-header">
                        <div className="notifications-popup-header-title">
                            9 New
                        </div>
                        <div className="notifications-popup-header-subtitle">
                            User notifications
                        </div>
                    </div>
                </div>
            </FadeInContainer>
        )
    }

    render() {
        const { showPopup } = this.state;

        return (
            <HideOnClickOutsideContainer
                onHide={() => this.togglePopup(false)}
            >
                <div className="notifications">
                    <button
                        onClick={this.onNotificationsClick}
                        className="notification-button"
                    >
                        <i className="fa far fa-bell" />
                    </button>
                    <div className="notifications-animation-wrapper">
                        <TransitionGroup>
                            {showPopup && this.renderNotifications()}
                        </TransitionGroup>
                    </div>
                </div>
            </HideOnClickOutsideContainer>
        );
    }
}
