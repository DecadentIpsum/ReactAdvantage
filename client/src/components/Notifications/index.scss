@import "global-styles/common.scss";

.notification-button {
    background: none;
    border: none;
    cursor: pointer;
    font-size: 1rem;
    color: $primary;
}

.notifications {
    position: relative;

    @media (max-width: 640px) {
        position: static;
    }
}

.notifications-animation-wrapper {
    position: absolute;
    top: 45px;
    left: 50%;
    transform: translate(-50%, 0);

    @media (max-width: 640px) {
        left: 5vw;
        transform: none;
    }
}

.notifications-popup {
    width: 400px;
    padding: 20px;
    box-sizing: border-box;
    min-height: 400px;
    background: $white;
    border-radius: 3px;
    box-shadow: 0 0 15px 1px rgba(81,77,92,.2);

    @media (max-width: 640px) {
        width: 90vw;
    }

    &::after {
        content: "";
        position: absolute;
        bottom: 100%;
        left: 50%;
        height: 0;
        width: 0;
        border: solid transparent;
        pointer-events: none;
        border-color: rgba($white, 0);
        border-bottom-color: $white;
        border-width: 10px;
        margin-left: -10px;

        @media (max-width: 640px) {
            display: none;
        }
    }
}

.notifications-popup-header {
    padding-bottom: 15px;
    border-bottom: 1px solid rgba($grayShade, 0.5);
    text-align: center;
}

.notifications-popup-header-title {
    margin-bottom: 10px;
    font-size: 1.3rem;
    color: $blackShade;
}

.notifications-popup-header-subtitle {
    font-size: 1rem;
    color: $grayShade;
}