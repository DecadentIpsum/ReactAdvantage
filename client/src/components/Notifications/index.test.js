import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Notifications from './index';

it('renders correctly', () => {
    const component = shallow(<Notifications />);

    expect(component).toMatchSnapshot();
});

describe('popup is showing after click on trigger button', () => {
    const component = shallow(<Notifications />);
    const button = component.find('.notification-button');

    test('popup is hidden by default', () => {
        expect(component.state().showPopup).toBe(false);
    });

    test('popup is visible after click on button', () => {
        button.simulate('click');
        expect(component.state().showPopup).toBe(true);
    });
});
