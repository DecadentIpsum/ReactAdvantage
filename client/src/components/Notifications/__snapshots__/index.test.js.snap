// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notifications />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="notifications"
>
        <button
                className="notification-button"
                onClick={[Function]}
        >
                <i
                        className="fa far fa-bell"
                />
        </button>
        <div
                className="notifications-animation-wrapper"
        >
                <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                />
        </div>
</div>,
      "onHide": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="notification-button"
            onClick={[Function]}
>
            <i
                        className="fa far fa-bell"
            />
</button>,
          <div
            className="notifications-animation-wrapper"
>
            <TransitionGroup
                        childFactory={[Function]}
                        component="div"
            />
</div>,
        ],
        "className": "notifications",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa far fa-bell"
/>,
            "className": "notification-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa far fa-bell",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TransitionGroup
              childFactory={[Function]}
              component="div"
/>,
            "className": "notifications-animation-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "childFactory": [Function],
              "children": false,
              "component": "div",
            },
            "ref": null,
            "rendered": false,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="notifications"
>
          <button
                    className="notification-button"
                    onClick={[Function]}
          >
                    <i
                              className="fa far fa-bell"
                    />
          </button>
          <div
                    className="notifications-animation-wrapper"
          >
                    <TransitionGroup
                              childFactory={[Function]}
                              component="div"
                    />
          </div>
</div>,
        "onHide": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="notification-button"
              onClick={[Function]}
>
              <i
                            className="fa far fa-bell"
              />
</button>,
            <div
              className="notifications-animation-wrapper"
>
              <TransitionGroup
                            childFactory={[Function]}
                            component="div"
              />
</div>,
          ],
          "className": "notifications",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa far fa-bell"
/>,
              "className": "notification-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa far fa-bell",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TransitionGroup
                childFactory={[Function]}
                component="div"
/>,
              "className": "notifications-animation-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "childFactory": [Function],
                "children": false,
                "component": "div",
              },
              "ref": null,
              "rendered": false,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
